src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -2104,29 +2010,24 @@ private static MarshalAsAttribute GetMarshalAsCustomAttribute(RuntimeFieldInfo f
 
         private static MarshalAsAttribute GetMarshalAsCustomAttribute(int token, RuntimeModule scope)
         {
-            UnmanagedType unmanagedType, arraySubType;
-            VarEnum safeArraySubType;
-            int sizeParamIndex = 0, sizeConst = 0;
-            string marshalTypeName = null, marshalCookie = null, safeArrayUserDefinedTypeName = null;
-            int iidParamIndex = 0;
```

Comment:
Was this just defensive before? I did take a look at the native MetaDataImport::GetMarshalAs and it may throw an exception, but otherwise does initialize these out variables.

Reply:
Older coding style I assume, declaring and initialising variables at top of method.

If `MetaDataImport::GetMarshalAs` threw an exception its going pass straight out of this method and all these locals will be ignored anyway as they aren't out params; and it wouldn't be caught here.

Reply:
Aside: declaring and initialising the locals early is bad for register assignment; as it uses registers up when it doesn't have to